airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/operation/executor/DefaultCheckOperationExecutor.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.core.operation.executor
+
+import io.airbyte.cdk.core.context.env.ConnectorConfigurationPropertySource
+import io.airbyte.protocol.models.v0.AirbyteConnectionStatus
+import io.airbyte.protocol.models.v0.AirbyteMessage
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.micronaut.context.annotation.Requires
+import jakarta.inject.Named
+import jakarta.inject.Singleton
+
+private val logger = KotlinLogging.logger {}
+
+@Singleton
+@Named("checkOperationExecutor")
+@Requires(
+    property = ConnectorConfigurationPropertySource.CONNECTOR_OPERATION,
+    value = "check",
+)
+class DefaultCheckOperationExecutor : OperationExecutor {
```

Comment:
Why do we need both an Operation and an OperationExecutor? I'm sure there's a reason, but I don't get it.

Reply:
@postamar I attempted to answer this in an earlier comment, but it was a first pass/attempt at providing the appropriate logic split to allow for connectors to override the different behavior without needing to copy logic from the CDK.  For example, if you look at the CHECK operation, there is really two layers of logic:  the general logic that runs the check and if any exceptions are raised, returns a failure AND there is the actual check logic that talks to the source/destination system.  Most connectors will just want to override/replace the latter without having to touch the general logic.  There may be a few cases where the exception handling logic still needs to return success or something, so these two are separate out to facilitate that.  

That being said, I think that there may be three distinct operation types instead of one, so I am open to taking another pass at this as we implement some connectors using this new CDK.

Reply:
Got it, thanks!